homeassistant/components/volvooncall.py

```diff
@@ -148,29 +157,41 @@ def vehicle_name(self, vehicle):
 class VolvoEntity(Entity):
     """Base class for all VOC entities."""
 
-    def __init__(self, hass, vin, attribute):
+    def __init__(self, hass, vin, component, attribute):
         """Initialize the entity."""
-        self._hass = hass
-        self._vin = vin
-        self._attribute = attribute
-        self._state.entities[self._vin].append(self)
+        self.hass = hass
+        self.vin = vin
+        self.component = component
+        self.attribute = attribute
+        self.data.entities.append(self)
+
+    @property
+    def data(self):
+        """Return the shared state."""
+        return self.hass.data[DATA_KEY]
```

Comment:
Store the data instance on the entity as an attribute instead. Pass the data in from the platform via `hass.data` when creating the entity.

Reply:
✓